app/models/notification.rb

```diff
@@ -105,4 +126,20 @@ def web_url
   def repo_url
     "#{Octobox.github_domain}/#{repository_full_name}"
   end
+
+  def unarchive_if_updated
+    return unless self.archived?
+    change = changes['updated_at']
+    return unless change
+    if self.archived && change[1] > change[0]
+      self.archived = false
+    end
+  end
+
+  def update_from_api_response(api_response, unarchive: false)
+    attrs = Notification.attributes_from_api_response(api_response)
+    self.attributes = attrs
+    unarchive_if_updated if unarchive
+    save(touch: false) if changed?
```

Comment:
It would be nice to have a separate `github_updated_at` field so that we don't end up accidentally tainting `updated_at` by saving without `touch: false`